fabric-enchantment-api-v1/src/main/java/net/fabricmc/fabric/api/enchantment/v1/FabricEnchantment.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * <p>Please contact the author, Vaerian, at the email or GitHub profile listed above
+ * with any questions surrounding implementation choices, functionality, or updating
+ * to newer versions of the game.</p>
+ */
+public abstract class FabricEnchantment extends Enchantment {
+	protected FabricEnchantment(Rarity weight, EquipmentSlot...slotTypes) {
+		super(weight, EnchantmentTarget.ARMOR /*This can be anything, it just can't be null otherwise it'll cause a null pointer exception. At this point it has no bearing on functionality*/, slotTypes);
+	}
+
+	// Making this method override and abstract means it has to be overwritten by any children.
+	// By default vanilla defines this as simply delegating to the target, but obviously, because
+	// there is no target anymore it must be overridden in order to prevent a null pointer exception.
+	/**
+	 * Overrides the vanilla enchantment target and determines whether or not
+	 * the given item stack can accept this enchantment by any means.
+	 *
+	 * <p>Only this method is called directly by the anvil to determine whether or not
+	 * this enchantment can be applied to an item. Item stacks that do not indicate
+	 * enchantability ({@link net.minecraft.item.Item#isEnchantable(ItemStack)} and
+	 * {@link net.fabricmc.fabric.api.enchantment.v1.FabricEnchantment#isEnchantableItem(ItemStack)})
+	 * can still accept this enchantment through the anvil, but they will not be able to be enchanted
+	 * in the enchantment table itself</p>
+	 *
+	 * @param stack The item stack querying the ability to accept this
+	 * enchantment.
+	 * @return A boolean value representing whether or not the given item
+	 * stack should be able to accept the enchantment.
+	 *
+	 * @see net.minecraft.item.Item#isEnchantable(ItemStack)
+	 * @see net.minecraft.screen.EnchantmentScreenHandler#onContentChanged(Inventory)
+	 * @see net.fabricmc.fabric.mixin.enchantment.EnchantmentHelperMixin
+	 */
+	@Override
+	public abstract boolean isAcceptableItem(ItemStack stack);
+
+	/**
+	 * Overrides the vanilla judgement about whether or not the given item
+	 * stack can accept this enchantment from an enchantment table.
+	 *
+	 * <p>It is important to note that item stacks of items who do not say that
+	 * they are enchantable will not be able to be enchanted, despite this
+	 * method indicating that they can accept this enchantment. To change this
+	 * the {@link net.minecraft.item.Item#isEnchantable(ItemStack)} method
+	 * must be overridden.</p>
+	 *
+	 * @param stack The item stack querying the ability to accept this
+	 * enchantment.
+	 * @return A boolean value representing whether or not the given item
+	 * stack should be able to accept the enchantment from an enchantment
+	 * table.
+	 *
+	 * @see net.minecraft.item.Item#isEnchantable(ItemStack)
+	 * @see net.minecraft.screen.EnchantmentScreenHandler#onContentChanged(Inventory)
+	 * @see net.fabricmc.fabric.mixin.enchantment.EnchantmentHelperMixin
```

Comment:
Uh, don't link to mixins. They are good reference for impl detail but it hurts javadoc badly